lib/elixir/lib/process.ex

```diff
@@ -940,6 +940,25 @@ defmodule Process do
   @spec unalias(alias) :: boolean
   defdelegate unalias(alias), to: :erlang
 
+  @doc """
+  Add a descriptive term to the current process.
+
+  The term does not need to be unique, and in Erlang/OTP 27+ will be shown in Observer.
```

Comment:
- Even the use of the word `term` is already explaining it, I like how the [original doc](https://erlang.org/documentation/doc-15.0-rc1/lib/stdlib-6.0/doc/html/proc_lib.html#set_label/1) calls out explicitly that unlike the unregistered name it doesn't have to be an atom
- Not only in Observer, but also in crash logs (and possibly other debugging tools)

Reply:
- It's true that the label doesn't have to be an atom: I've tried strings, tuples, and maps, and all show up in Observer just fine. But I wonder if we should also say that an atom label would be the most memory efficient option?
- As I [described here](https://erlangforums.com/t/erlang-otp-27-0-rc1-released/3300/25?u=nathanl), I tried setting a label in a GenServer and crashing it, but didn't see it in the raised error. Maybe I should be looking elsewhere, but if we're going to say "it shows in crash logs" maybe we should clarify where to see that? (I'd need help there.)

Reply:
> But I wonder if we should also say that an atom label would be the most memory efficient option?

I wouldn't say so. The main goal of the label is to be descriptive and have stuff like "user-id-#{...}" for a LiveView process for example. I think not having it tied to atoms is the best idea to push forward.

> As I [described here](https://erlangforums.com/t/erlang-otp-27-0-rc1-released/3300/25?u=nathanl), I tried setting a label in a GenServer and crashing it, but didn't see it in the raised error.

We will need to update Elixir and other behaviours but it will be there by the time it is out, so it is safe to include it (and if it is not there, it is an Elixir bug).